core/src/main/scala/com/softwaremill/kmq/redelivery/streams/StartMarkerStream.scala

```diff
@@ -0,0 +1,38 @@
+package com.softwaremill.kmq.redelivery.streams
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.kafka.ConsumerMessage.CommittableMessage
+import akka.kafka.scaladsl.Consumer.DrainingControl
+import akka.kafka.scaladsl.{Committer, Consumer, Producer}
+import akka.kafka._
+import akka.stream.scaladsl.Sink
+import com.softwaremill.kmq.{MarkerKey, MarkerValue, StartMarker}
+import org.apache.kafka.clients.producer.ProducerRecord
+
+class StartMarkerStream(msgConsumerSettings: ConsumerSettings[String, String],
+                        markerProducerSettings: ProducerSettings[MarkerKey, MarkerValue],
+                        queueTopic: String, markersTopic: String, maxPartitions: Int, messageTimeout: Long)
+                       (implicit system: ActorSystem) {
+
+  def run(): DrainingControl[Done] = {
+    val committerSettings = CommitterSettings(system)
+
+    Consumer.committablePartitionedSource(msgConsumerSettings, Subscriptions.topics(queueTopic))
+      .mapAsyncUnordered(maxPartitions) {
+        case (topicPartition, source) =>
+          source
+            .map { msg: CommittableMessage[String, String] =>
+              ProducerMessage.single(
+                new ProducerRecord[MarkerKey, MarkerValue](markersTopic, MarkerKey.fromRecord(msg.record), new StartMarker(msg.record.timestamp() + messageTimeout)), //Note: handle redeliverAfter differently - save relative to message time
```

Comment:
I don't understand why `return new MarkerKey(r.partition(), r.offset());` contains partition and offset, this increases out-of-order processing, which is already lost with redeliveries, but why completely resign from it?

Reply:
After discussion: we agreed that out-of-order processing of messages in the queue is okay in this case.